fal/controllers/anime_stats.py

```diff
@@ -140,9 +143,24 @@ def populate_anime_weekly_stats() -> None:
     week = config.getint("weekly info", "current-week")
 
     with session_scope() as session:
-        anime_list = Season.get_season_from_database(
+        season_anime = Season.get_season_from_database(
             season_of_year, year, session).anime
 
+        anime_ids_collected = (row[0] for row in session.query(AnimeWeeklyStat.anime_id).filter(
+            AnimeWeeklyStat.week == week
+        ).all())
+
+        if anime_ids_collected:
```

Comment:
I'm assuming that this check is to make sure that `anime_ids_collected` isn't empty, but generator expressions return `True` even when it's empty. I tested this in the Python shell to make sure:

```bash
PS C:\Users\Abhinav> python
Python 3.7.2 (tags/v3.7.2:9a3ffc0492, Dec 23 2018, 22:20:52) [MSC v.1916 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> a = []
>>> x = (v for v in a)
>>> if x:
...     print('Hi')
...
Hi
```
So it might be better to use a list comprehension instead of a generator expression in this case.

Reply:
That explains why my test was failing!